### ISSUE TITLE
[Dark Theme] Checkbox created with toolkit on dialog has wrong background

### ISSUE THREAD

**Issue by marcushoepfner** [eclipse-platform/eclipse.platform.ui#416]
A button with SWT.CHECK gets a wrong background color (light grey instead of dark grey) under the following conditions:

- created on a dialog
- toolkit used (FormToolkit.createButton)
- darktheme

In an editor this does not happen.

<img width="153" alt="Screenshot 2022-10-17 at 13 30 50" src="https://user-images.githubusercontent.com/29301444/196166511-a59e5ed3-404c-47a6-b7fc-66ab3c514fc9.png">

Does happen on Mac. Not on Windows.

**Comment by akurtakov** [comment]
Can you share the snippet to reproduce?

**Comment by marcushoepfner** [comment]
sure.
[background.test_1.0.0.202210171431.zip](https://github.com/eclipse-platform/eclipse.platform.ui/files/9800462/background.test_1.0.0.202210171431.zip)

(It's actually a jar, but github does not support. You might need to rename back to .jar)

**Comment by marcushoepfner** [comment]
I have started analyzing this:
- FormToolkit calls adapt() when creating the checkbox; a background color is set
- afterwards the css engine overwrites this background color via Control.setBackground(Color color) with null
- this leads to the light grey eventually

Compared to e.g. a label created with toolkit
- FormToolkit calls adapt() when creating the label; a background color is set
-  afterwards the css engine overwrites this background color via Control.setBackground(Color color) with a concrete color

Is just the color in css missing?

**Comment by marcushoepfner** [comment]
Does someone has a hint how to fix this? If so, I can take over and try.

**Comment by akurtakov** [comment]
Can you share the code for this jar so it can be used for debugging?

**Comment by marcushoepfner** [comment]
Oh damn. I forgot. Thanks for bearing with me.
Here it is, same disclaimer: rename from zip to jar
[background.test.source_1.0.0.202210191250.zip](https://github.com/eclipse-platform/eclipse.platform.ui/files/9819767/background.test.source_1.0.0.202210191250.zip)

**Comment by marcushoepfner** [comment]
Hi, any progress on this one?

**Comment by akurtakov** [comment]
Unfortunately not many people look into MacOS specific issues (@Phillipus only lately) . In my opinion this could be SWT bug but it could be css styling issue too. In order to get this issue moving people with interest (and access) in MacOS have to step up but you can further help by trying to reduce the snippet to pure SWT snippet one .

**Comment by Phillipus** [comment]
> not sure whether it happens on Windows.

OK on Windows:

![Image 001](https://github.com/eclipse-platform/eclipse.platform.ui/assets/600504/533c698f-629b-4e8b-b990-01d3c5478939)

> reduce the snippet to pure SWT snippet one .

Not sure if that's possible as it depends on dark theme and Theming framework.

@marcushoepfner Perhaps you might investigate the dark *.css files in the [theme plug-in](https://github.com/eclipse-platform/eclipse.platform.ui/tree/master/bundles/org.eclipse.ui.themes/css) to see if any change required there?

**Comment by BeckerWdf** [comment]
I even see issues in the light theme on macOS:

<img width="182" alt="image" src="https://github.com/user-attachments/assets/95f81f2c-f910-4323-ae94-1eec7cfe56a4">

**Comment by marcushoepfner** [comment]
> Better Checkbox Coloring on Dialog when using FormToolkit #2474

Correct. I've seen this as well. There is also an issue with light theme.

**Comment by BeckerWdf** [comment]
> I even see issues in the light theme on macOS:
> 
> <img alt="image" width="182" src="https://private-user-images.githubusercontent.com/28338612/384308368-95f81f2c-f910-4323-ae94-1eec7cfe56a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNTg3OTcsIm5iZiI6MTczMTA1ODQ5NywicGF0aCI6Ii8yODMzODYxMi8zODQzMDgzNjgtOTVmODFmMmMtZjkxMC00MzIzLWFlOTQtMWVlYzdjZmU1NmE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDA5MzQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzNjczNmJmZTNmN2E5NjkyMDY4MWQ0OTU4NTcxYTk2MGFkMTE0NDNlNWYzMmNmNDYwMjU2OWUxZDIxNGE5Y2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zn7B8PMVbGdg9dUVpkxjmZsDVRrAuVMu8d6aDUUX2Qw">

It looks the same when theming is "turned off" in the preferences. The widget should always look correct also when theming is off. So fixing this via CSS feels wrong to me.
